ext/standard/file.c

```diff
@@ -1831,8 +1831,8 @@ PHP_FUNCTION(fputcsv)
 	if (delimiter_str != NULL) {
 		/* Make sure that there is at least one character in string */
 		if (delimiter_str_len < 1) {
-			php_error_docref(NULL, E_WARNING, "delimiter must be a character");
-			RETURN_FALSE;
+			zend_value_error("delimiter must be a character");
```

Comment:
(Drive-by) This message is somewhat confusing, maybe "delimiter cannot be empty"?

Reply:
That message would better reflect the check, but actually only the first `char` is ever used as delimiter; maybe it's worth introducing a pontetial BC break here, and check for `delimiter_str_len == 1)`? Same for `enclosure` a few lines below.

Reply:
Another possibility would be to actually add support for multiple chars (effectively: Unicode support).

Reply:
> 
> 
> Another possibility would be to actually add support for multiple chars (effectively: Unicode support).

Tried to have a look, it seems that the whole fgetcsv and fputcsv sub-routines need to be rewritten. I don't mind either way to merge the checks or have them adapted to have multibyte support. But I feel this should rather be in a separate PR.